test/Vegetation/plant_hydraulics_test.jl

```diff
@@ -0,0 +1,199 @@
+                root_depths,
+                z_stem_midpoint,
+                p_soil0,
+                Y[1],
+                vg_α,
+                vg_n,
+                vg_m,
+                ν,
+                S_s,
+                K_sat_root,
+                K_sat_stem,
+            ),
+        )
+        flux_out_stem = flux(
+            z_stem_midpoint,
+            z_leaf_midpoint,
+            Y[1],
+            Y[2],
+            vg_α,
+            vg_n,
+            vg_m,
+            ν,
+            S_s,
+            K_sat_stem,
+            K_sat_leaf,
+        )
+
+        F[1] = flux_in_stem - T0
+        F[2] = flux_out_stem - T0
+    end
+
+    soln = nlsolve(initial_rhs!, [-0.03, -0.02])
+    p_stem_0 = soln.zero[1]
+    p_leaf_0 = soln.zero[2]
+
+    S_l_stem_0 =
+        inverse_water_retention_curve(vg_α, vg_n, vg_m, p_stem_0, ν, S_s)
+    S_l_leaf_0 =
+        inverse_water_retention_curve(vg_α, vg_n, vg_m, p_leaf_0, ν, S_s)
+
+    ϑ_l_stem_0 = augmented_liquid_fraction(ν, S_l_stem_0)
+    ϑ_l_leaf_0 = augmented_liquid_fraction(ν, S_l_leaf_0)
+
+    ϑ_l_0 = [ϑ_l_stem_0, ϑ_l_leaf_0]
+    Y, p, coords = initialize(plant_hydraulics)
+
+    Y.vegetation.ϑ_l .= ϑ_l_0
+
+    plant_hydraulics_ode! = make_ode_function(plant_hydraulics)
+
+    hour = FT(60^2)
+    day = FT(24 * 60^2)
+    t0 = FT(0)
+    tf = FT(hour * 24 * 4)
+    dt = FT(1)
+
+    prob = ODEProblem(plant_hydraulics_ode!, Y, (t0, tf), p)
+    sol = solve(prob, Euler(), dt = dt)
+
+    dY = similar(Y)
+    plant_hydraulics_ode!(dY, Y, p, 0.0)
+    @test sqrt(mean(dY.vegetation.ϑ_l .^ 2.0)) < 1e-8 # starts in equilibrium
+
+    ϑ_l_stem = reduce(hcat, sol.u)[1, :]
+    ϑ_l_leaf = reduce(hcat, sol.u)[2, :]
+
+    S_l_stem = effective_saturation.(ν, ϑ_l_stem)
+    S_l_leaf = effective_saturation.(ν, ϑ_l_leaf)
+
+    p_stem = water_retention_curve.(vg_α, vg_n, vg_m, S_l_stem, ν, S_s)
+    p_leaf = water_retention_curve.(vg_α, vg_n, vg_m, S_l_leaf, ν, S_s)
+
+    function final_rhs!(F, Y)
+        p_soilf = p_soil0
+        Tf = FT(1e-5)
+        flux_in_stem = sum(
+            flux.(
+                root_depths,
+                z_stem_midpoint,
+                p_soilf,
+                Y[1],
+                vg_α,
+                vg_n,
+                vg_m,
+                ν,
+                S_s,
+                K_sat_root,
+                K_sat_stem,
+            ),
+        )
+        flux_out_stem = flux(
+            z_stem_midpoint,
+            z_leaf_midpoint,
+            Y[1],
+            Y[2],
+            vg_α,
+            vg_n,
+            vg_m,
+            ν,
+            S_s,
+            K_sat_stem,
+            K_sat_leaf,
+        )
+        F[1] = flux_in_stem - Tf
+        F[2] = flux_out_stem - Tf
+    end
+
+    # Check that the final state is in the new equilibrium
+    soln = nlsolve(final_rhs!, [-0.03, -0.02]; ftol = 1e-10)
+    p_stem_f = soln.zero[1]
+    p_leaf_f = soln.zero[2]
+    @test abs(p_stem_f - p_stem[end]) < 0.1
```

Comment:
is this a high enough tolerance? 

Maybe it is enough that the system starts in equilibrium (line 147) and we do not need to do the integration as well.

Reply:
agreed, removed it.